examples/coreset_in_model_pipeline.py

```diff
@@ -0,0 +1,250 @@
+# pylint: disable=too-many-statements
+# pylint: disable=duplicate-code
+def main(
+    out_path: Optional[Path] = None,
+) -> tuple[tuple[float, float], tuple[float, float], tuple[float, float]]:
+    """
+    Run machine learning pipeline example.
+
+    A 1-dimensional set of features is generated from the standard Gaussian
+    distribution a set of paired responses is generated with a non-linear relationship
+    to the features. A coreset is then generated using Kernel Herding and a Kernel Ridge
+    Regression model is fit using it. It is then compared in terms of time to fit the
+    model and test mean squared error to the full dataset and a random subset.
+
+    :param out_path: Path to save output to, if not :data:`None`, assumed relative to
+        this module file unless an absolute path is given
+    :return: ((full data test MSE, fitting time), (coreset test MSE, fitting time),
+        (random test MSE, fitting time))
+    """
+    # Create some data. Here we'll use 25,000 points in 1D from a uniform distribution
+    seed = 1_234
+    num_data_pairs = 25_000
+    feature_key, error_key = jr.split(jr.key(seed), 2)
+    x = jr.normal(feature_key, shape=(num_data_pairs, 1))
+
+    def _generating_func(x: Array) -> Array:
+        """Encode non-linear relationship between features and response."""
+        return (
+            x
+            + x**2
+            + 5 * jnp.sin(x)
+            + 100 * jnp.exp(-((x - 3) ** 2))
+            + 100 * jnp.exp(-((x + 2) ** 2))
+        )
+
+    y = _generating_func(x) + jr.normal(error_key, shape=(num_data_pairs, 1))
+    x, x_test, y, y_test = train_test_split(x, y, test_size=0.1, random_state=seed)
+
+    # Scale the features and responses individually
+    feature_scaler = StandardScaler().fit(x)
+    x = jnp.asarray(feature_scaler.transform(x))
+    x_test = jnp.asarray(feature_scaler.transform(x_test))
+
+    response_scaler = StandardScaler().fit(y)
+    y = jnp.asarray(response_scaler.transform(y))
+    y_test = jnp.asarray(response_scaler.transform(y_test))
+
+    # Request 200 coreset points
+    coreset_size = 200
+
+    # Setup the original data object
+    data = Data(x)
+
+    # Set the bandwidth parameter of the kernel using a median heuristic derived from at
+    # most 1000 random samples in the data.
+    num_samples_length_scale = min(num_data_pairs, 1_000)
+    generator = np.random.default_rng(seed)
+    idx = generator.choice(num_data_pairs, num_samples_length_scale, replace=False)
+    length_scale = median_heuristic(x[idx]).item()
+    kernel = SquaredExponentialKernel(length_scale=length_scale)
+
+    print("Estimating kernel ridge regression model with full data...")
+    full_data_fit_time = time()
+    full_data_model = KernelRidge(kernel="rbf", gamma=length_scale, alpha=1e-2).fit(
+        x, y
+    )
+    full_data_fit_time = time() - full_data_fit_time
+    full_data_mse = jnp.linalg.norm(y_test - full_data_model.predict(x_test))
```

Comment:
I missed this the first time round, but this isn't the mean squared error, it's the square root of the sum of squared errors. (This is 1.875 on my machine, which is clearly incorrect looking at the graph!) The same issue is present for the other two "MSE" variables.

Reply:
(I'd prefer that the statistic displayed on the graph was the root mean squared error, as that's more readily interpretable than the mean squared error.)

Reply:
Good spot, will fix